worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/FactoryConverter.java

```diff
@@ -42,29 +49,56 @@
 import org.enginehub.piston.inject.Key;
 
 import java.util.List;
+import java.util.function.Consumer;
 import java.util.function.Function;
 
 public class FactoryConverter<T> implements ArgumentConverter<T> {
 
+    @AutoAnnotation
+    private static ClipboardMask clipboardMask() {
+        return new AutoAnnotation_FactoryConverter_clipboardMask();
+    }
+
     public static void register(WorldEdit worldEdit, CommandManager commandManager) {
         commandManager.registerConverter(Key.of(Pattern.class),
-            new FactoryConverter<>(worldEdit, WorldEdit::getPatternFactory, "pattern"));
+            new FactoryConverter<>(worldEdit, WorldEdit::getPatternFactory, "pattern", c -> {}));
         commandManager.registerConverter(Key.of(Mask.class),
-            new FactoryConverter<>(worldEdit, WorldEdit::getMaskFactory, "mask"));
+            new FactoryConverter<>(worldEdit, WorldEdit::getMaskFactory, "mask", c -> {}));
         commandManager.registerConverter(Key.of(BaseItem.class),
-            new FactoryConverter<>(worldEdit, WorldEdit::getItemFactory, "item"));
+            new FactoryConverter<>(worldEdit, WorldEdit::getItemFactory, "item", c -> {}));
+
+        commandManager.registerConverter(Key.of(Mask.class, clipboardMask()),
+                new FactoryConverter<>(worldEdit, WorldEdit::getMaskFactory, "mask",
+                        context -> {
+                            try {
+                                ClipboardHolder holder = context.getSession().getClipboard();
+                                Transform transform = holder.getTransform();
+                                Extent target;
+                                if (transform.isIdentity()) {
+                                    target = holder.getClipboard();
+                                } else {
+                                    target = new BlockTransformExtent(holder.getClipboard(), transform);
+                                }
+                                context.setExtent(target);
+                            } catch (EmptyClipboardException e) {
+                                throw new IllegalStateException(e);
+                            }
+                        }));
     }
 
     private final WorldEdit worldEdit;
     private final Function<WorldEdit, AbstractFactory<T>> factoryExtractor;
     private final String description;
+    private final Consumer<ParserContext> contextTweaker;
 
     private FactoryConverter(WorldEdit worldEdit,
                              Function<WorldEdit, AbstractFactory<T>> factoryExtractor,
-                             String description) {
+                             String description,
+                             Consumer<ParserContext> contextTweaker) {
```

Comment:
Might be nicer to allow `null` for this, and just pass in `null` instead of `c -> {}`.